cachi2/core/package_managers/yarn/main.py

```diff
@@ -126,7 +126,10 @@ def _configure_yarn_version(project: Project) -> None:
         project.package_json["packageManager"]
     )
 
-    if yarn_path_version is None and package_manager_version is None:
+    version = yarn_path_version if yarn_path_version else package_manager_version
+
+    # this check is done here to make mypy understand that version can't be Optional anymore
+    if version is None:
```

Comment:
Hold on, so NOW mypy understood? I tried a couple of compound if clause variations and it still complained, so this is the way? Stupid mypy.